packages/nextjs/src/server/__tests__/clerkMiddleware.test.ts

```diff
@@ -226,6 +226,33 @@ describe('clerkMiddleware(params)', () => {
     expect(decryptedData).toEqual(options);
   });
 
+  it('allows access to request object to dynamically define options', async () => {
```

Comment:
Will also try to add a test for the usage of `clerkClient` when passing keys with that option callback.